smart-jump-intero.el

```diff
@@ -0,0 +1,40 @@
+;;; smart-jump-intero.el --- Register `intero-mode' for `smart-jump'. -*- lexical-binding: t -*-
+
+;; Copyright (C) 2017 James Nguyen, Terje Larsen
+
+;; Author: James Nguyen <james@jojojames.com>, Terje Larsen <terlar@gmail.com>
+;; Maintainer: James Nguyen <james@jojojames.com>, Terje Larsen <terlar@gmail.com>
+;; URL: https://github.com/jojojames/smart-jump
+;; Version: 0.0.1
+;; Package-Requires: ((emacs "25.1"))
+;; Keywords: emacs, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;;; Register `intero-mode' for `smart-jump'.
+
+;;; Code:
+(require 'smart-jump)
+(require 'intero nil t)
+
+(defun smart-jump-intero-register ()
+  "Register `intero-mode' for `smart-jump'."
+  (smart-jump-register :modes 'intero-mode
+                       :jump-fn 'intero-goto-definition
+                       :pop-fn 'xref-pop-marker-stack
+                       :refs-fn 'smart-jump-simple-find-references))
```

Comment:
I don't think this exists anymore. I removed it as part of pushing to melpa.

smart-jump-find-references-with-ag is what's left but I kind of consider that internal. Can you add that function back (smart-jump-simple-find-references) and just have it call smart-jump-find-references-with-ag?

I can do that too at some point once I'm free.

Reply:
Yes, I prefer to have it back, as then people can override it if they prefer some other function for finding references and can still utilize the default configurations.

I did that for example, I used:
```
(defun +smart-jump-find-references-with-counsel-rg ()
  "Use `rg' and `counsel' to find references."
  (interactive)
  (if (fboundp 'counsel-rg)
      (counsel-rg
       (cond ((use-region-p)
              (buffer-substring-no-properties (region-beginning)
                                              (region-end)))
             ((symbol-at-point)
              (substring-no-properties
               (symbol-name (symbol-at-point))))))
    (message "Install swiper to use `+smart-jump-simple-find-references-with-counsel-rg'.")))
```

Reply:
This might be good to add to smart-jump too and make it part of the defcustom for smart-jump-simple-find-references.